### DISCUSSION TITLE
Browser event is not being dispatched on dehydrate

### DISCUSSION THREAD

**Discussion by mira-thakkar** [livewire/livewire#5015] [1. Help]
Laravel : "^8.75"
Livewire:"^2.10"

I want to implement custom selectbox in my component, so wanted do it using dispatch browser event on dehydrate(as it's going to be called after every render.

Component File
```
public function dehydrate()
{
        logger()->info('hydrate');
        $this->dispatchBrowserEvent('select-box');
}
```
Component View File
```
<section>
  --- Component HTML-----
 <script>
  window.addEventListener('select-box', function () {
     console.log('hi');
     alert("hi");
  });
 </script>
</section>
```
The strange thing is not getting this alert or console from JS. I am not able to figure out what is going wrong here. Any help would be appreciated!
Thanks!

**Comment by joshhanley** [comment]
@mira-thakkar Livewire can't dispatch browser events on the initial request as Livewire's javascript object hasn't been loaded yet. So that would be why you're not seeing the alert on page load.

But if you triggered another request (using `$refresh` or something), then you should see the event.

Hope this helps!
